### PR TITLE
[#176654017] Add secrets management

### PR DESCRIPTION
**Pull request by pasqualedevita**
This PR add secrets management reading values from azure keyvault.

for **pagopa-github-bot-ro-TOKEN** (read-only) create a personal access token with this permissions:
- admin:repo_hook, read:user, user:email

for **pagopa-github-bot-rw-TOKEN** (read-write) create a personal access token with this permissions:
- read:user, repo, user:email

for **pagopa-github-bot-pr-TOKEN** (pull request) create a personal access token with this permissions:
- public_repo

Update TOKEN values into azure keyvault

### CODE REVIEW

File: azure-devops/projects/io-backend-projects/secrets.tf [hunk 1]
```diff
@@ -0,0 +1,56 @@
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_key_vault" "keyvault" {
+  name                = "io-p-kv-azuredevops"
+  resource_group_name = "io-p-rg-operations"
+}
+
```
Comment (uolter) [inline azure-devops/projects/io-backend-projects/secrets.tf:9]: I will replace all these data blocks with something like:

variable "secrets" {
  default = [
    "DANGER-GITHUB-API-TOKEN",
    "pagopa-github-bot-ro-TOKEN",
    "pagopa-github-bot-rw-TOKEN",
    "pagopa-github-bot-pr-TOKEN",
    "pagopa-github-bot-EMAIL",
    "pagopa-github-bot-USERNAME",
    "TTDIO-PROD-IO-SUBSCRIPTION-ID",
    "TTDIO-DEV-IO-SUBSCRIPTION-ID",
    "TTDIO-SPN-TENANTID",
  ]
}


data "azurerm_key_vault_secret" "key_vault_secret" {
  for_each = var.secrets
  name         = each.value
  key_vault_id = data.azurerm_key_vault.keyvault.id
}

Do not repeat yourself and less is more   :-)

**Review by uolter — COMMENTED** [review]
What are they for all these data?
I mean you are queering many secrets here but you are not using them. Am I wrong?

File: azure-devops/projects/io-backend-projects/secrets.tf (hunk 1 above)
Comment (pasqualedevita) [reply azure-devops/projects/io-backend-projects/secrets.tf:23]: updated, thank you @uolter

**Comment by pasqualedevita** [comment]
> What are they for all these data?
> I mean you are queering many secrets here but you are not using them. Am I wrong?

Yes, in next pr I'll start to use them

**Review by uolter — APPROVED** [review]

**Pull request merged**